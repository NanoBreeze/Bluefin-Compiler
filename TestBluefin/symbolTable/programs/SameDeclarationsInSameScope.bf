struct First {
	int b;
	int b ;
	int c;
	float c;
	bool d() {}
	void d;
} ;

First f(float a) {
	First a;
	{ 
		int b;
		string b;
	}
}

First a;
string b;

bool f(int x, float x) {}
int First;