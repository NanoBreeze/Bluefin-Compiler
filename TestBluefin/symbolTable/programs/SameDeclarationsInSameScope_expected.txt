declare - First - c_struct - t_First
setCurrentScope - Level 1
resolve - int - c_builtinType - t_int
declare - b - c_var - t_int
resolve - int - c_builtinType - t_int
declare - b - REDECLARATION
resolve - int - c_builtinType - t_int
declare - c - c_var - t_int
resolve - float - c_builtinType - t_float
declare - c - REDECLARATION
resolve - bool - c_builtinType - t_bool
declare - d - c_function - t_bool
enterScope - Level 2
exitScope - Level 2
resolve - void - c_builtinType - t_void
declare - d - REDECLARATION
exitScope - Level 1
resolve - First - c_struct - t_First
declare - f - c_function - t_First
enterScope - Level 1
resolve - float - c_builtinType - t_float
declare - a - c_var - t_float
resolve - First - c_struct - t_First
declare - a - REDECLARATION
enterScope - Level 2
resolve - int - c_builtinType - t_int
declare - b - c_var - t_int
resolve - string - c_builtinType - t_string
declare - b - REDECLARATION
exitScope - Level 2
exitScope - Level 1
resolve - First - c_struct - t_First
declare - a - c_var - t_First
resolve - string - c_builtinType - t_string
declare - b - c_var - t_string
resolve - bool - c_builtinType - t_bool
declare - f - REDECLARATION
enterScope - Level 1
resolve - int - c_builtinType - t_int
declare - x - c_var - t_int
resolve - float - c_builtinType - t_float
declare - x - REDECLARATION
exitScope - Level 1
resolve - int - c_builtinType - t_int
declare - First - REDECLARATION
